src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -95,7 +95,7 @@ public async virtual Task<HostSecretsInfo> GetHostSecretsAsync()
                         // so we read the secrets running them through the appropriate readers
                         hostSecrets = ReadHostSecrets(hostSecrets);
                     }
-                    catch (CryptographicException)
+                    catch (CryptographicException ex) when (!ex.InnerException.IsFatal())
```

Comment:
No null ref concern with the `IsFatal()` call?

Reply:
The `IsFatal` handles `null` parameters and iterates over the inner exceptions, so it's safe here.